pkg/engine/deployment.go

```diff
@@ -343,3 +341,22 @@ func assertSeen(seen map[resource.URN]deploy.Step, step deploy.Step) {
 func isDefaultProviderStep(step deploy.Step) bool {
 	return providers.IsDefaultProvider(step.URN())
 }
+
+func checkTargets(targetUrns deploy.UrnTargets, snap *deploy.Snapshot) error {
```

Comment:
Be nice to fold this back down into the engine but I can do that after this merges